### DISCUSSION TITLE
Refreshing tokens & time limits

### DISCUSSION THREAD

**Discussion by briangru** [timdorr/tesla-api#562] [Q&A]
When refreshing multiple tokens, are there any time limits on how frequently we can update a batch?  The goal here is to not overwhelm Tesla's servers, and also not get flagged by their web application firewall.  

The rough guidance from multiple threads in 2021 was that when logging in, wait 60 seconds between login attempts to avoid being throttled and/or banned.  I'm looking for similar guidance on using refresh tokens to get new access tokens.  Should we assume the same, or can we use something like 1 second instead?

**Comment by core-hacked** [comment]
I suggest you take the token's expiry time ``8 hours`` and update when it's almost expired, which will avoid any flagging or throttling. Otherwise just refresh at least 5 mins apart, as that will not overwhelm tesla's servers as much. 

> You should be able to refresh anytime, but there is no point in refreshing so much.
